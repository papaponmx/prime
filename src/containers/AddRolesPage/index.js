import React, { Component } from 'react';
import { connect } from 'react-redux';
import Snackbar from '@material-ui/core/Snackbar';

import AddRoleForm from '../../components/AddRoleForm';
import {closeSnackbar} from '../../actions';

class AddRolesPage extends Component {
    render() {
      const open = (this.props.snackbar && this.props.snackbar.open)
      ? this.props.snackbar.open
      : false;
      const message = (this.props.snackbar && this.props.snackbar.message)
      ? this.props.snackbar.message
      : '';
        return (
            <div>
              <h1>Add Roles</h1>
              <AddRoleForm />
              <Snackbar
                open={open}
                message={message}
                autoHideDuration={4000}
                onRequestClose={this.props.closeSnack}
              />
            </div>
        )
    }
}

const mapStateToProps= state => ({
  snackbar: state.roles.snackbar,
});

const mapDispatchToProps = (dispatch)=> ({
  closeSnack: () => dispatch(closeSnackbar())
});


export default connect(mapStateToProps, mapDispatchToProps)(AddRolesPage);
