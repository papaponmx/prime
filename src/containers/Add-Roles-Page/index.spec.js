import React from 'react';
import AddRolesPage from '.';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

describe('AddRolesPage', () => {
  const output = shallow(
    <AddRolesPage />
  );
  const header = output.find('h1');
  it('Sould Render add roles page', () => {  
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Should have the Add Roles header', () => {  
    expect(header.text()).toMatch(/Add Roles/);
  });
});
