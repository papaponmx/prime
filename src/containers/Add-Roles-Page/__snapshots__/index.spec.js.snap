// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRolesPage Sould Render add roles page 1`] = `
<div>
  <h1>
    Add Roles
  </h1>
  <Connect(ReduxForm) />
</div>
`;
